TensorFI/faultTypes.py

```diff
@@ -174,3 +174,78 @@ def bitTensor ( dtype, val):
 	val = val.reshape(valShape)
 	return dtype.type( val )
 
+def bitMultiScalar( dtype, val ):
+	"Flip multiple bits of the scalar value"   
+	fiConf = injectFault.getFIConfig()
+	
+	def getBinary(number):
+		# integer data type
+		if(floor(number) == number):
+			integer = bin(int(number)).lstrip("0b") 
+			# 21 digits for integer
+			integer = integer.zfill(21)
+			# integer has no mantissa
+			dec = ''	
+		# float point datatype 						
+		else:
+			binVal = float2bin(number)				
+			# split data into integer and decimal part	
+			integer, dec = binVal.split(".")	
+		return integer, dec
+
+	# we use a tag for the sign of negative val, and then consider all values as positive values
+	# the sign bit will be tagged back when finishing bit flip
+	negTag = 1
+	if(str(val)[0]=="-"):
+		negTag=-1
+
+	if(isinstance(val, np.bool_)):	
+		# boolean value
+		return bool( (val+1)%2 )
+	else:	
+		# turn the val into positive val
+		val = abs(val)
+		integer, dec = getBinary(val)
+
+	intLength = len(integer)
+	decLength = len(dec)
+
+	# random indices of the bits to flip  
+	list = np.random.sample(range(intLength + decLength), fiConf.bitCount)
+	
+	# iterate through the list
+	for index in list:
+		# flip the sign bit (optional)
```

Comment:
We consider sign bit also now, there's no reason not to, check recent commit by Zitao and update this as well